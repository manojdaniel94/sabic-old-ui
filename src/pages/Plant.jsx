import React, { useRef } from "react";
import AffiliatesBreadCrumb from "../Components/AffiliatesPage/AffiliatesBreadCrumb";
//import Status from "../Components/PlantComponent/Status";
import Status from "../Components/Assets/Status/Status";
import Affiliates from "../Components/PlantComponent/Affiliates";
import HealthIndex from "../Components/PlantComponent/HealthIndex";
import BreadCrumb from "../Components/BreadCrumbComponent/BreadCrumb";


const Plant = () => {
    return (
        <>
            <BreadCrumb />
            <div id="content-affiliates">
                <Status />
                <Affiliates />
                <HealthIndex />
            </div>
        </>

    );
};

export default Plant;