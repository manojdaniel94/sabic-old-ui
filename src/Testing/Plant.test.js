import React from "react";
import { render } from "@testing-library/react";
import Plant from "../pages/Plant";

describe("Plant component", () => {
  test("Plant component renders the AffiliatesBreadCrumb component", () => {
    const { getByText } = render(<Plant />);
    const text = getByText("Affiliates");
    expect(text).toBeInTheDocument();
  });

  it("renders the Status component", () => {
    const { getByTestId } = render(<Plant />);
    const status = getByTestId("status");
    expect(status).toBeInTheDocument();
  });

  it("renders the Affiliates component", () => {
    const { getByTestId } = render(<Plant />);
    const affiliates = getByTestId("affiliates");
    expect(affiliates).toBeInTheDocument();
  });

  it("renders the HealthIndex component", () => {
    const { getByTestId } = render(<Plant />);
    const healthIndex = getByTestId("health-index");
    expect(healthIndex).toBeInTheDocument();
  });
});
