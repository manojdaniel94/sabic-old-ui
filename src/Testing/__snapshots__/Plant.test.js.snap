// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plant component renders without crashing 1`] = `
<DocumentFragment>
  <div
    id="content-breadcrumb"
  >
    <a
      href="#"
    >
       SABIC
    </a>
    <a
      href="#"
    >
      All Regions
    </a>
    <a
      class="active"
      href="#"
    >
      affiliates
    </a>
  </div>
  <div
    id="content-affiliates"
  >
    <div
      class="affiliates-outer"
    >
      <div
        id="status"
      >
        <div
          class="common1"
        >
          <div
            class="c-affilate"
          >
            <span
              id="value"
            >
              19
            </span>
            <span>
              AFFILIATES
            </span>
          </div>
          <div
            class="c-assets"
          >
            <span>
              464
            </span>
            <span>
              ASSETS
            </span>
          </div>
          <div
            class="c-healthindex"
          >
            <span>
              23 %
            </span>
            <span>
              HEALTH INDEX
            </span>
          </div>
          <div
            class="c-compliance"
          >
            <span>
              16
            </span>
            <span>
              ALERTS
            </span>
          </div>
          <div
            class="c-compliance"
          >
            <span>
              16
            </span>
            <span>
              IN PROGRESS
            </span>
          </div>
          <div
            class="c-compliance"
          >
            <span>
              0
            </span>
            <span>
              OVERDUE
            </span>
          </div>
          <div
            class="c-compliance"
          >
            <span>
              0
            </span>
            <span>
              CLOSED
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="affiliates-outer"
    >
      <div
        id="affiliates"
      >
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
        <div
          class="affiliates-box"
        >
          <div>
            OLF 1
          </div>
          <div>
            <span>
              HEALTH INDEX
            </span>
            <span
              class="green"
            >
              29%
            </span>
          </div>
          <div>
            <span>
              PM COMPLIANCE
            </span>
            <span
              class="blue"
            >
              70%
            </span>
          </div>
          <div>
            <span>
              ACTIVE ALERTS
            </span>
            <span
              class="red"
            >
              87
            </span>
          </div>
          <div>
            <span>
              UNDER INVESTIGATION
            </span>
            <span
              class="blue"
            >
              1
            </span>
          </div>
          <div>
            <span>
              OVERDUE INVESTIGATION
            </span>
            <span
              class="red"
            >
              86
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="affiliates-outer"
    >
      <h3>
        HEALTH INDEX
      </h3>
      <div
        data-testid="health-index"
      >
        <div
          class="health-affiliate"
        />
        <div
          class="health-affiliate"
        />
        <div
          class="health-affiliate"
        >
          <div
            class="health-box"
          >
            <div
              class="bar"
              style="height: 120px;"
            >
              <span>
                8
              </span>
            </div>
          </div>
          <div
            class="health-af-name"
          >
            OLF 1
          </div>
        </div>
        <div
          class="health-affiliate"
        >
          <div
            class="health-box"
          >
            <div
              class="bar"
              style="height: 80px;"
            >
              <span>
                5
              </span>
            </div>
          </div>
          <div
            class="health-af-name"
          >
            OLF 2
          </div>
        </div>
        <div
          class="health-affiliate"
        >
          <div
            class="health-box"
          >
            <div
              class="bar"
              style="height: 40px;"
            >
              <span>
                2
              </span>
            </div>
          </div>
          <div
            class="health-af-name"
          >
            OLF 3
          </div>
        </div>
        <div
          class="health-affiliate"
        />
        <div
          class="health-affiliate"
        />
        <div
          class="health-affiliate"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
